File: company/CPSCO/2019/session3/D2.py
N = int( input())
S = input()
ans = "Yes"
T = [0]*N
T[0] = 1
if S[0] != "R":
    ans = "No"
else:
    for i in range(1,N):
        if T[i-1] == 1:
            if S[i] == "R":
                T[i] = 1
            elif S[i] == "G":
                T[i] = 2
            else:
                ans = "No"
                break
        elif T[i-1] == 2:
            if S[i] == "R":
                T[i] = 1
            elif S[i] == "B":
                T[i] = 3
            else:
                ans = "No"
                break
        elif T[i-1] == 3:
            if S[i] == "R":
                T[i] = 1
            elif S[i] == "G":
                T[i] = 4
            else:
                T[i] = 3
        elif T[i-1] == 4:
            if S[i] == "R":
                T[i] = 1
            elif S[i] == "B":
                T[i] = 5
            else:
                ans = "No"
                break
        elif T[i-1] == 5:
            if S[i] == "R":
                T[i] = 1
            elif S[i] == "G":
                T[i] = 4
            else:
                T[i] = 5

if T[-1] == 1 or T[-1] == 2 or T[-1] == 4:
    ans = "No"
print(ans)

File: rust/abc143/src/bin/d2.rs
// -*- coding:utf-8-unix -*-
#![allow(non_snake_case)]

#[cfg(debug_assertions)]
#[allow(unused)]
macro_rules! eprintln {
    ($p:tt, $($x:expr),*) => {
        std::eprintln!($p, $($x,)*);
    };
}

#[cfg(not(debug_assertions))]
#[allow(unused)]
macro_rules! eprintln {
    ($p:tt, $($x:expr),*) => {};
}

use proconio::{fastout, input};
// use proconio::marker::Bytes;
// use proconio::marker::Usize1;

#[fastout]
fn main() {
    input!{
        N: usize,
        mut L: [usize; N],
    }
    let mut ans = 0;
    L.sort();
    for i in 0..(N-2){
        for j in (i+1)..(N-1){
            for k in (j+1)..N{
                if L[k] < L[i] + L[j]{
                    ans += 1;
                }
            }
        }
    }
    
    println!("{}", ans);
}
